### DISCUSSION TITLE
File filter

### DISCUSSION THREAD

**Discussion by XlebniKomok** [rejetto/hfs#259] [Q&A]
Hello. In the old version of the program there was a "file filter". How is it implemented now?

**Comment by rejetto** [comment]
it's still possible, but a bit hard because I need to do more work on it.

But while answering I noticed something is not working as I expected, so I'll let you know.

**Comment by rejetto** [comment]
If you want to forbid mp3 files, you can enter this in the "masks" field
```
"*.mp3": 
  can_read: false
```

If you want to allow only mp3 files, you can enter this in the "masks" field
```
"*.mp3": 
  can_read: true
"*": 
  can_read: false
```

As I said, I will see to make this easier.

**Reply by XlebniKomok** [reply]
How to apply this to subfolders?

**Reply by rejetto** [reply]
the shortest is to use **.mp3 instead of *.mp3

**Reply by rejetto** [reply]
i see there is a limit with my suggestion, and folders are not handled as one would want.
For the moment the workaround is to use `"*.*"` instead of `"*"`, but it's no perfect solution and I need to take care of this.

**Reply by XlebniKomok** [reply]
`"*.*"` does not work `"**.*"` work but `"*.mp3": can_read: true` and `"**.mp3": can_read: true ` stops working, For now, it's enough for me to hide certain types of files. Will there be a feature to hide empty folders like in the old version?

**Reply by rejetto** [reply]
do you need the "hide empty folders" option?

**Reply by XlebniKomok** [reply]
In some cases, yes, but not so important. If it does not notice a lot of time, you can add.

**Reply by rejetto** [reply]
ok, i will consider it at some point, but have other priorities right now

**Comment by rejetto** [comment]
ok, i started a new page to talk about this
https://github.com/rejetto/hfs/wiki/Permissions
